glean-core/src/lib.rs

```diff
@@ -732,6 +732,27 @@ impl Glean {
         self.debug.debug_view_tag.get()
     }
 
+    /// Set source tags.
+    ///
+    /// This will return `false` in case `value` contains invalid tags.
+    ///
+    /// Ping tags will show in the destination datasets, after ingestion.
+    ///
+    /// ## Arguments
+    ///
+    /// * `value` - A vector of at most 5 valid HTTP header values. Must match the regex: "[a-zA-Z0-9-]{1,20}".
+    pub fn set_source_tags(&mut self, value: Vec<String>) -> bool {
+        self.debug.sourge_tags.set(value)
+    }
+
+    /// Return the value for the source tags or `None` if it hasn't been set.
+    ///
+    /// The source_tags may be set from an environment variable (GLEAN_SOURCE_TAGS)
+    /// or through the `set_source_tags` function.
+    pub(crate) fn source_tags(&self) -> Option<&Vec<String>> {
+        self.debug.sourge_tags.get()
```

Comment:
`&Vec<...>` is a weird type to see. Usually one would expect a `&[...]`

Since 1.40.0 there's [`as_deref`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref) on `Option`, so I think this should do the trick:
```suggestion
    pub(crate) fn source_tags(&self) -> Option<&[String]> {
        self.debug.sourge_tags.get().as_deref()
```

Reply:
Your code is not wrong though, so I leave it to you to decide if that additional call is worth it.

Reply:
Oh, TIL! Unfortunately this also means I should change `DebugOption`, otherwise the compiler is sad with

>  expected slice, found struct `std::vec::Vec`